app/models/tenant.rb

```diff
@@ -240,6 +240,12 @@ def any_editable_domains?
     ae_domains.where(:system => false).count > 0
   end
 
+  def reset_domain_priority_by_ordered_ids(ids)
+    uneditable_domains = visible_domains - editable_domains
+    uneditable_domains.delete_if { |domain| domain.name == 'ManageIQ' }
```

Comment:
@gmcculloug 
MiqAeDomain doesn't have tenant information so it won't know what was displayed to the user. The tenant is the one that knows about the visible and editable domains.